src/common/intrusive_lru.h

```diff
@@ -236,7 +254,11 @@ void intrusive_ptr_release(intrusive_lru_base<Config> *p) {
   assert(p->use_count > 0);
   --p->use_count;
   if (p->use_count == 0) {
-    p->lru->mark_as_unreferenced(*p);
+    if (p->is_referenced()) {
+      p->lru->mark_as_unreferenced(*p, true);
+    } else {
+      delete p;
```

Comment:
ditto: This branch suggests that last reference was released but the element was already marked as un referenced.